sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/export/__init__.py

```diff
@@ -0,0 +1,195 @@
+        self._telemetry_processors = []
+
+    def _apply_telemetry_processors(
+        self, envelopes: typing.List[TelemetryItem]
+    ) -> typing.List[TelemetryItem]:
+        """Applies all telemetry processors in the order they were added.
+
+        This function will return the list of envelopes to be exported after
+        each processor has been run sequentially. Individual processors can
+        throw exceptions and fail, but the applying of all telemetry processors
+        will proceed (not fast fail). Processors also return True if envelope
+        should be included for exporting, False otherwise.
+
+        Args:
+            envelopes: The envelopes to apply each processor to.
+        """
+        filtered_envelopes = []
+        for envelope in envelopes:
+            accepted = True
+            for processor in self._telemetry_processors:
+                try:
+                    if processor(envelope) is False:
+                        accepted = False
+                        break
+                except Exception as ex:
+                    logger.warning("Telemetry processor failed with: %s.", ex)
+            if accepted:
+                filtered_envelopes.append(envelope)
+        return filtered_envelopes
+
+    def _transmit_from_storage(self) -> None:
+        for blob in self.storage.gets():
+            # give a few more seconds for blob lease operation
+            # to reduce the chance of race (for perf consideration)
+            if blob.lease(self.options.timeout + 5):
+                envelopes = blob.get()
+                result = self._transmit(envelopes)
+                if result == ExportResult.FAILED_RETRYABLE:
+                    blob.lease(1)
+                else:
+                    blob.delete()
+
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-nested-blocks
+    # pylint: disable=too-many-return-statements
+    def _transmit(self, envelopes: typing.List[TelemetryItem]) -> ExportResult:
+        """
+        Transmit the data envelopes to the ingestion service.
+
+        Returns an ExportResult, this function should never
+        throw an exception.
+        """
+        if len(envelopes) > 0:
+            try:
+                track_response = self.client.track(envelopes)
+                if not track_response.errors:
+                    logger.info("Transmission succeeded: Item received: %s. Items accepted: %s",
+                                track_response.items_received, track_response.items_accepted)
+                    return ExportResult.SUCCESS
+                else:
+                    resend_envelopes = []
+                    for error in track_response.errors:
+                        if self._is_retryable_code(error.statusCode):
+                            resend_envelopes.append(
+                                envelopes[error.index]
+                            )
+                        else:
+                            logger.error(
```

Comment:
do we want to raise here?

Reply:
I don't believe we want to ever crash the user's application during runtime.